rq/cli/helpers.py

```diff
@@ -198,6 +201,54 @@ def setup_loghandlers_from_args(verbose, quiet, date_format, log_format):
     setup_loghandlers(level, date_format=date_format, log_format=log_format)
 
 
+def job_func(func, args, kwargs):
+    try:
+        importlib.import_module(func)  # execute file
+    except ImportError:
+        return import_attribute(func)(*args, **kwargs)
+
+
+def parse_function_arg(argument):
+    if argument.startswith(':'):  # no keyword, json
+        return None, loads(argument[1:])
```

Comment:
Can we use [literal_eval()](https://docs.python.org/3/library/ast.html#ast.literal_eval) instead of JSON to evaluate the arguments?

It's better this way since this is native to Python so we can do:
```
{"key": True}
```
Instead of converting it to JSON like such:
```
{"key": True}
```

I know I suggested we look to httpie for inspiration, but they chose JSON because it's native to how HTTP works.

Reply:
> Can we use [literal_eval()](https://docs.python.org/3/library/ast.html#ast.literal_eval) instead of JSON to evaluate the arguments?

Like I mentioned [here](https://github.com/rq/rq/pull/1466#discussion_r651275396) I would prefer this updated version of [this comment](https://github.com/rq/rq/pull/1466#discussion_r650510889) (Not implemented yet):

This would support both `json.loads` and `ast.literal_eval`.

| | keyword | no keyword |
|-|-|-|
| string | `[key]=[value]` | `[value]`<br>Should not be used if you can't ensure, that the string doesn't contains `:`, `=` or `@`. |
| json | `[key]:=[value]` | `:[value]` |
| ast.literal_eval | `[key]#=[value]` | `#[value]` |

Please write me if I souldn't implement this.

> 
> It's better this way since this is native to Python so we can do:
> 
> [...]
> 
> I know I suggested we look to httpie for inspiration, but they chose JSON because it's native to how HTTP works.

That's true, but if you call this command from another programm, because it is not written in python, I don't think there's any python-eval serializer, but there are json serializers in many programming languages.

Reply:
I changed `#` to `%` because # will start a comment in .